inject_test.go

```diff
@@ -257,6 +270,30 @@ func TestInject(t *testing.T) {
 		assert.Nil(t, out.Result.T2, "T2 must be nil")
 		assert.True(t, gave1 == out.Result.T1, "T1 must match")
 	})
+
+	t.Run("FieldsCanBeOptional", func(t *testing.T) {
```

Comment:
Could you add 2 more tests:
1. For private embedded fields.
2. For a recursive dig.In struct(for my curiousity):
```golang
type A struct{
  fx.In
  B struct {
  fx.In
  
  C  int
  }
}
```